### PR TITLE
Rename executable and releases to `Arduino Cloud Agent`

### LINKED ISSUE #931: Rename the "Arduino Create Agent" to "Arduino Cloud Agent" (phase 1)
The current name of the Arduino Agent software should be modified to be in line with the name of the Arduino cloud offerings, that is "Arduino Cloud" (instead of Arduino Create).

This change is motivated by two reasons:
- Improve clarity and reduce misunderstandings (users should not get confused by naming differences). Note that most of the Arduino documentation, tutorials, and blogs reference this software as the "Arduino Cloud Agent".
- Promote the new Arduino branding (Cloud instead of Create).

An internal analysis was conducted to find most of the things that need to be changed. This task focuses on the first sub-set of changes that are either more user-visible or easier to perform. For example changes involving user visible strings or files. Other things like names inside the source code, configuration files and other more complex changes (installer) can be postponed.

List of things to change (not definitive yet):
- Change the name of the application executable (or .app folder)
- Change the name of the installer file and inside the InstallBuilder installer (for platforms that use the installer)
- Update the "Debug Console"
- Update the README and other documentation in the repository

**Note**: the rename activities may affect the Agent update process, as it tries to replace itself. This must be considered when updating from a version with the old name to a version with the new name.

### PR DESCRIPTION
**Pull request by MatteoPologruto**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-create-agent/pulls)
      before creating one)
- [ ] Tests for the changes have been added (for bug fixes / features)

* **What kind of change does this PR introduce?**

The current name of the Arduino Agent software should be modified to be in line with the name of the Arduino cloud offerings, that is "Arduino Cloud" (instead of Arduino Create).  List of things that have been modified:
- Change the name of the application executable (or .app folder)
- Update the "Debug Console"
- Update the README and other documentation in the repository

- **Does this PR introduce a breaking change?**

No

* **Other information**:

Test release: https://github.com/arduino/arduino-create-agent/releases/tag/untagged-26461045e3dbac347ae4
PR on the Agent Installer repo: https://github.com/bcmi-labs/arduino-create-agent-installer/pull/28

### CODE REVIEW

File: home.html
Comment (umbynos) [inline home.html:1]: I like the new linting/formatting but it would have been more clear in a different commit (for the future)

**Review by umbynos — COMMENTED** [review]
Could you try to run a test release?
Let's remember to update the wiki documentation too after this change.
Do we need to update the installbuilder configuration?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -300,7 +300,7 @@ jobs:
         run: gon -log-level=debug -log-json "${{ env.GON_CONFIG_PATH }}"
 
       - name: Zip output app bundle
-        run: ditto -c -k ArduinoCreateAgent.app/ ArduinoCreateAgent.app_${{ matrix.arch }}_notarized.zip
+        run: ditto -c -k ArduinoCreateAgent.app/ ArduinoCloudAgent.app_${{ matrix.arch }}_notarized.zip
```
Comment (umbynos) [inline .github/workflows/release.yml:303]:
```suggestion
        run: ditto -c -k ArduinoCloudAgent.app/ ArduinoCloudAgent.app_${{ matrix.arch }}_notarized.zip
```
Shouldn't update this too? Along with L289

**Review by umbynos — APPROVED** [review]
Tested on linux and works as expected, nice work

**Review by Xayton — APPROVED** [review]

**Pull request merged**